sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -75,7 +77,7 @@
    */
   protected long totalRowCount;
 
-  protected ParquetFileReader reader;
+  protected ParquetRowGroupReader reader;
```

Comment:
IMO. This should be emphasized in the title instead of adding more test coverage. :)